src/bridge/IrcBridge.ts

```diff
@@ -802,9 +801,9 @@ export class IrcBridge {
                 // all the clients already connected to avoid races.
                 syncer.sync().then(() =>
                     discoveringClientsPromise
-                ).finally(() => {
+                ).finally(() =>
```

Comment:
I'll test this myself, but may as well ask now: does awaiting on `promise.finally(callback)` await on the original `promise` (meaning the callback is always resolved asynchronously), or on the Promise returned by the callback (as long as it's non-rejected)?

Reply:
`.finally` awaits on the promise it's a member of. I.e. in this case `discoveringClientsPromise`